src/huggingface_hub/hf_api.py

```diff
@@ -1861,7 +1878,11 @@ def upload_file(
                 raise err
 
         d = r.json()
-        return d["url"]
+
+        if "error" in d:
+            logger.error(d["error"])
```

Comment:
Not sure what this response looks like in the case of an error...should this be raising something as well?